app/controllers/registrations_controller.rb

```diff
@@ -0,0 +1,30 @@
+class RegistrationsController < Devise::RegistrationsController
+  def create
+    build_resource(sign_up_params)
+
+    if resource.save
+      if resource.active_for_authentication?
+        set_flash_message :notice, :signed_up if is_navigational_format?
```

Comment:
from 7-9 and and 11-13 almost the same code.

Reply:
It is.